pilot/pkg/bootstrap/server.go

```diff
@@ -80,6 +80,8 @@ const (
 	CopilotTimeout = 5 * time.Second
 	// FilepathWalkInterval dictates how often the file system is walked for config
 	FilepathWalkInterval = 100 * time.Millisecond
+	// ClusterRegistriesDir defines a directory to store kubeconf read from ConfigMap
+	ClusterRegistriesDir = "/tmp/kubeconf"
```

Comment:
Why do we need to store it ? Just add a comment - it's a temp file, so I don't think it's a huge problem, but I assumed you can initialize kubeclient from an in-memory struct.

Reply:
BTW - can you make it a flag ? If all secrets are stored in a single file, we can just mount the secret.

Reply:
@costinm that was my initial intention to use in memory struct, but then you guys are creating clients all over the place and use kubeconf file in many cases. I decided to go through the path of less resistance and instead of refactoring all references to kubeconf file, just to follow the flow and add other cluster memebers kubeconf to the folder from where the rest of the code could access it.

Reply:
I think if we can get to in-memory representation of the secrets, that is more secure.  The in-cluster config will still contain the in-cluster token (in /var/lib/wherever it goes) however, the less leakage of secrets into the control plane filesystem space the better.

Reply:
Not sure I understand. In-memory is good - but /tmp/kubeconf is not in-memory.

Keep in mind pilot may run as process on raw VMs, not only in k8s containers - and having secrets in /tmp/ is dangerous. At least '/var/lib is supposed to be owned by a istio user. If you use tmp - make sure permissions are set...

What I don't get is: if you end up with secrets on a file, why not mount it ? This seems the worse of allworlds - still file, hardcoded in tmp and more complex.